src/module/iotjs_module_buffer.c

```diff
@@ -119,6 +119,33 @@ static size_t bound_range(int index, size_t low, size_t upper) {
 }
 
 
+static unsigned hex2bin(char c) {
+  if (c >= '0' && c <= '9')
+    return c - '0';
+  if (c >= 'A' && c <= 'F')
+    return 10 + (c - 'A');
+  if (c >= 'a' && c <= 'f')
+    return 10 + (c - 'a');
+  return (unsigned)(-1);
```

Comment:
I added a debug assert. 
And `0` means success. I think, `-1` is suitable in this point.